2_gitlab/create-repos.sh

```diff
@@ -13,34 +13,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ -z ${GITLAB_HOSTNAME} ];then
-  read -p "What is the GitLab hostname (i.e. my.gitlab.server)? " GITLAB_HOSTNAME
-fi
-
-if [ -z ${GITLAB_TOKEN} ];then
-read -s -p "What is the access token? " GITLAB_TOKEN
-fi
-
 REPOS="anthos-config-management shared-kustomize-bases shared-ci-cd kustomize-docker kaniko-docker golang-template golang-template-env java-template java-template-env"
 CLUSTERS="prod-us-central1 prod-us-east1 staging-us-central1"
+# Create SSH keys so ACM syncers can read from the repos
+# at base of 2_gitlab
+export WORKINGDIR=$(pwd)
+mkdir -p ${WORKINGDIR}/ssh-keys
+
 pushd gitlab-repos
-  # Create SSH keys so ACM syncers can read from the repos
-  mkdir -p ../../ssh-keys
-  pushd ../../ssh-keys
+  # TODO: Convert this into Terraform using Google Cloud KMS
+  pushd ${WORKINGDIR}/ssh-keys
     for repo in ${REPOS}; do
        test -f ${repo} || ssh-keygen -f ${repo} -N ''
     done
     for cluster in ${CLUSTERS}; do
        test -f ${cluster} || ssh-keygen -f ${cluster} -N ''
+       # Create a secret with the private key
+       gcloud secrets create gitlab-cluster-key-${cluster} --replication-policy=automatic --data-file <(cat "${cluster}")
     done
   popd
+
   terraform init
-  terraform plan -var gitlab_token=${GITLAB_TOKEN} -var gitlab_hostname=${GITLAB_HOSTNAME}
-  if [ -z ${TERRAFORM_AUTO_APPROVE} ]; then
-    terraform apply -var gitlab_token=${GITLAB_TOKEN} -var gitlab_hostname=${GITLAB_HOSTNAME}
-  else
-    terraform apply -auto-approve -var gitlab_token=${GITLAB_TOKEN} -var gitlab_hostname=${GITLAB_HOSTNAME}
-  fi
+  terraform plan -var gitlab_token=${GITLAB_TOKEN} -var gitlab_hostname=${GITLAB_HOSTNAME} -var ssh-key-path-base=${WORKINGDIR}/ssh-keys -out=terraform.tfplan
+  terraform apply -auto-approve terraform.tfplan
+
+  # TODO: Move into terraform sometime in the future, for now, forcefully delete in destroy
+  gcloud secrets create gitlab-user --replication-policy=automatic --data-file <(echo -n "root")
```

Comment:
Should add a note to the README about the secrets being there for more persistent storage.

Reply:
Added